wis2box-management/wis2box/downloader.py

```diff
@@ -0,0 +1,113 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+###############################################################################
+
+import logging
+
+import click
+import requests
+
+from wis2box import cli_helpers
+
+LOGGER = logging.getLogger(__name__)
+
+DOWNLOAD_URL = 'http://wis2downloader:5000'
+
+
+@click.group()
+def downloader():
+    """Interact with the wis2downloader"""
+    pass
+
+
+@click.command('list-subscriptions')
+@click.pass_context
+@cli_helpers.OPTION_VERBOSITY
+def list_subscriptions(ctx, verbosity):
+    """list all subscriptions"""
+
+    # make a GET requests to http://{DOWNLOAD_URL}/subscriptions
+    try:
+        response = requests.get(f'{DOWNLOAD_URL}/subscriptions')
+        # check response status
+        if response.status_code == 200:
+            click.echo('Current subscriptions:')
+            click.echo(response.text)
+        else:
+            click.echo(f'Error: {response.status_code}')
+            click.echo(response.text)
+    except requests.exceptions.ConnectionError:
+        click.echo('Error: Connection refused')
+        click.echo(f'Is the wis2downloader running on {DOWNLOAD_URL}?')
+
+
+@click.command('add-subscription')
+@click.pass_context
+@click.option('--topic', '-t', help='The topic to subscribe to', required=True)
+def add_subscription(ctx, topic):
+    """add a subscription"""
+
+    topic = topic.replace('#', '%23')
+    topic = topic.replace('+', '%2B')
+    # make a POST request to http://{DOWNLOAD_URL}/subscriptions
+    try:
+        response = requests.post(f'{DOWNLOAD_URL}/subscriptions?topic={topic}')
+        # check response status
+        if response.status_code == 200:
+            click.echo('Subscription added')
+            click.echo('Current subscriptions:')
+            click.echo(response.text)
+        else:
+            click.echo('Subscription not added')
+            click.echo(f'Error: {response.status_code}')
+            click.echo(response.text)
+    except requests.exceptions.ConnectionError:
+        click.echo('Error: Connection refused')
+        click.echo(f'Is the wis2downloader running on {DOWNLOAD_URL}?')
+
+
+@click.command('remove-subscription')
+@click.pass_context
+@click.option('--topic', '-t', help='The topic to subscribe to', required=True)
+def remove_subscription(ctx, topic):
+    """remove a subscription"""
+
+    topic = topic.replace('#', '%23')
+    topic = topic.replace('+', '%2B')
+    # make a DELETE request to http://{DOWNLOAD_URL}/subscriptions
+    try:
+        response = requests.delete(f'{DOWNLOAD_URL}/subscriptions?topic={topic}')  # noqa
```

Comment:
The topic is now given as part of the url, e.g. 

requests.delete(f'{DOWNLOAD_URL}/subscriptions/cache/a/wis2/%2B/core/%23')